src/main/java/com/exactpro/th2/common/schema/factory/CommonFactory.java

```diff
@@ -465,10 +466,23 @@ public Set<String> getDictionaryAliases() {
             }
 
             try (Stream<Path> files = Files.list(dictionaryFolder)) {
```

Comment:
suggestion: This is completely up to you but you can write the same a bit shorter (and clearer, in my opinion)

```java
            try (Stream<Path> files = Files.list(dictionaryFolder)) {
                Function<Path, String> getAlias = path -> FilenameUtils.removeExtension(path.getFileName().toString()).toLowerCase();
                Map<String, Set<Path>> filesByAlias = files.filter(Files::isRegularFile)
                        .collect(Collectors.groupingBy(getAlias, Collectors.toSet()));
                Map<String, Set<Path>> duplicates = filesByAlias.entrySet().stream()
                        .filter(entry -> entry.getValue().size() > 1)
                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
                if (!duplicates.isEmpty()) {
                    throw new IllegalStateException("Dictionary directory contains files with the same name in different cases, files: " + duplicates + ", path: " + dictionaryFolder.toAbsolutePath());
                }
                return Set.copyOf(filesByAlias.keySet());
            }
```